lib/memo_wise.rb

```diff
@@ -94,6 +94,20 @@ def inherited(subclass)
   end
   private_constant(:CreateMemoWiseStateOnInherited)
 
+  module CreateMemoWiseStateOnIncluded
+    def included(base)
+      return unless base.is_a?(Class) && !base.singleton_class?
+
+      base.prepend(Module.new do
+        def initialize(...)
```

Comment:
I think this will need to change to support earlier Ruby versions, but just keeping it as-is for a proof-of-concept.